### PR TITLE
Fixed mount command to bind on specific VM ip address instead of all …

### PR DESCRIPTION
**Pull request by aaron-prindle**
…interfaces

### CODE REVIEW

**Comment by aaron-prindle** [comment]
Fixes #1383

**Comment by aaron-prindle** [comment]
@minikube-bot retest this please

**Comment by aaron-prindle** [comment]
This seems to be working, I believe the Linux-Virtualbox test is a flake as that is how I tested it locally

File: cmd/minikube/cmd/mount.go [hunk 1]
```diff
@@ -79,13 +81,23 @@ var mountCmd = &cobra.Command{
 			os.Exit(1)
 		}
 		defer api.Close()
+		host, err := api.Load(config.GetMachineName())
+		if err != nil {
+			glog.Errorln("Error getting IP: ", err)
```
Comment (dlorenc) [inline cmd/minikube/cmd/mount.go]: Wrong error message?

File: pkg/minikube/cluster/cluster_windows.go [hunk 2]
```diff
@@ -41,14 +40,18 @@ func createHypervHost(config MachineConfig) drivers.Driver {
 	return d
 }
 
-func getVMHostIP(host *host.Host) (net.IP, error) {
+func GetVMHostIP(host *host.Host) (net.IP, error) {
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster_windows.go]: Do you think we could move this up to the main cluster.go file and drop the _windows, _darwin variants? The switch/case should be able to handle this by itself.

Comment (aaron-prindle) [inline pkg/minikube/cluster/cluster_windows.go]: done

File: cmd/minikube/cmd/mount.go (hunk 1 above)
Comment (aaron-prindle) [inline cmd/minikube/cmd/mount.go]: done

File: pkg/minikube/cluster/cluster.go [hunk 3]
```diff
@@ -421,6 +422,51 @@ func MountHost(api libmachine.API, path string) error {
...
 }
 
+// GetVMHostIP gets the ip address to be used for mapping host -> VM and VM -> host
+func GetVMHostIP(host *host.Host) (net.IP, error) {
+	switch host.DriverName {
+	case "kvm":
+		return net.ParseIP("192.168.42.1"), nil
+	case "hyperv":
+		re := regexp.MustCompile("\"VSwitch\": \"(.*?)\",")
+		hypervVirtualSwitch := re.FindStringSubmatch(string(host.RawDriver))[1]
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster.go]: Can you leave a TODO here to deserialize the driver?

File: pkg/minikube/cluster/cluster.go [hunk 4]
```diff
@@ -421,6 +422,51 @@ func MountHost(api libmachine.API, path string) error {
...
+		ip, err := getIPForInterface(fmt.Sprintf("vEthernet (%s)", hypervVirtualSwitch))
+		if err != nil {
+			return []byte{}, errors.Wrap(err, "Error getting 9p mount command")
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster.go]: Should we switch up these error messages too? It's more about an error getting the IP.

File: pkg/minikube/cluster/cluster.go [hunk 5]
```diff
@@ -421,6 +422,51 @@ func MountHost(api libmachine.API, path string) error {
...
+		}
+		return ip, nil
+	case "xhyve":
+		return net.ParseIP("192.168.64.1"), nil
+	default:
+		return []byte{}, errors.New("Error, attempted to get host ip address for unsupported driver")
+	}
+}
+
+func getIPForInterface(name string) (net.IP, error) {
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster.go]: Did you get this from somewhere? Can you include a pointer to a doc or something?

File: pkg/minikube/cluster/cluster.go (hunk 3 above)
Comment (aaron-prindle) [inline pkg/minikube/cluster/cluster.go]: done

File: pkg/minikube/cluster/cluster.go (hunk 4 above)
Comment (aaron-prindle) [inline pkg/minikube/cluster/cluster.go]: done

File: pkg/minikube/cluster/cluster.go (hunk 5 above)
Comment (aaron-prindle) [inline pkg/minikube/cluster/cluster.go]: done

**Comment by sampowers** [comment]
I have checked, with xhyve at least, that this build's 'minikube mount' command only binds on the necessary address.  Thank your for your prompt fix!

File: pkg/minikube/cluster/cluster.go [hunk 6]
```diff
@@ -421,6 +422,55 @@ func MountHost(api libmachine.API, path string) error {
...
+		return []byte{}, errors.New("Error, attempted to get host ip address for unsupported driver")
+	}
+}
+
+// Based on code from http://stackoverflow.com/questions/23529663/how-to-get-all-addresses-and-masks-from-local-interfaces-in-go
+func getIPForInterface(name string) (net.IP, error) {
+	i, _ := net.InterfaceByName(name)
+	addrs, _ := i.Addrs()
+	for _, a := range addrs {
+		switch a.(type) {
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster.go]: Would a type assertion work here instead?

```golang
for _, a := range addrs {
  if ip, ok := a.(net.IPNet); ok {
  ...
  }
}
```

File: pkg/minikube/cluster/cluster.go [hunk 7]
```diff
@@ -421,6 +422,55 @@ func MountHost(api libmachine.API, path string) error {
...
+		case *net.IPNet:
+			ip := a.String()
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster.go]: Could you access the net.IPNet field `IP`, then use To4 to check if it's v4?
https://golang.org/pkg/net/#IP.To4

Then I think you could avoid stripping the CIDR as well.

File: pkg/minikube/cluster/cluster.go (hunk 6 above)
Comment (aaron-prindle) [inline pkg/minikube/cluster/cluster.go]: done

File: pkg/minikube/cluster/cluster.go (hunk 7 above)
Comment (aaron-prindle) [inline pkg/minikube/cluster/cluster.go]: done

File: pkg/minikube/cluster/cluster.go [hunk 8]
```diff
@@ -421,6 +422,47 @@ func MountHost(api libmachine.API, path string) error {
...
+		re := regexp.MustCompile("\"VSwitch\": \"(.*?)\",")
+		// TODO(aprindle) Change this to deserialize the driver instead
+		hypervVirtualSwitch := re.FindStringSubmatch(string(host.RawDriver))[1]
+		ip, err := getIPForInterface(fmt.Sprintf("vEthernet (%s)", hypervVirtualSwitch))
+		if err != nil {
+			return []byte{}, errors.Wrap(err, "Error getting VM/Host IP address")
+		}
+		return ip, nil
+	case "virtualbox":
+		ip, err := getIPForInterface("vboxnet0")
```
Comment (tbp105) [inline pkg/minikube/cluster/cluster.go]: The virtualbox network is not always vboxnet0.  If you had a vboxnet0 before you installed minikube, you'll end up with vboxnet1 (or higher)

**Pull request merged**